### ISSUE TITLE
RLS: 1.2.5

### ISSUE THREAD

**Issue by simonjayhawkins** [pandas-dev/pandas#40917]
Tracking issue for the 1.2.5 release.

https://github.com/pandas-dev/pandas/milestone/85

Currently scheduled for May 10, 2021

List of open regressions: https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression

**Comment by xmatthias** [comment]
Any chance of having #40505 backported to 1.2.5?

it's fixing a regression that ocured between 1.1.5 and 1.2.0, so i think this should be considered for a backport.

**Comment by jreback** [comment]
@xmatthias no that would be a too subtle change for a patch release

**Comment by xmatthias** [comment]
@jreback considering that it was an **undocumented** regression / behaviour change between 1.1.5 and 1.2.0, i tend to disagree.

**Comment by phofl** [comment]
This does fix a regression yes, but it also changes the behavior in certain cases, so agree with @jreback here, to subtle to include in 1.2.5

**Comment by simonjayhawkins** [comment]
tl;dr lmk if any blockers @pandas-dev/pandas-core @pandas-dev/pandas-triage 

---

it's probably be easier if we release 1.2.5 (if we are going to have another patch release in 1.2.x) before 1.3.0rc0.

so should probably think about releasing early next week.

lmk if any blockers, if we do another release on 1.2.x, I expect this to be the last in the series, so unlike the other patch releases that were calendar based, we want to make sure we include anything that should be included so that we don't need to do another.

We only have 2 patches, 1 of which was not trivial, so we could also decide not to do another patch release on 1.2.x

There's 1 open PR for a regression but's that's for an undocumented usage so could decide not to fix.

**Comment by jreback** [comment]
prob ok to release as we have right now

**Comment by simonjayhawkins** [comment]
we have a few outstanding issues but no PRs to fix. https://github.com/pandas-dev/pandas/milestone/85 except #41806

so could maybe do release after that one.

I've not yet got round to look at #41797 in detail, but looking at the diff of the PR that caused the regression https://github.com/pandas-dev/pandas/pull/37355/files, any fix is probably backportable.

also not sure if all the regressions in 1.2.x have been milestoned. could take the time to go back over the open issues.

This is hopefully the last in the 1.2.x series, so any delay to fix the outstanding issues maybe worthwhile.

Ideally want to release 1.2.5 before 1.3.0rc but can workaround if not.

**Comment by jreback** [comment]
the 1.2.x should be orthogonal to the 1.3.x i think

but yeah let's just release 1.2.5 when ready

**Comment by simonjayhawkins** [comment]
> the 1.2.x should be orthogonal to the 1.3.x i think

if we do it after, we need to backport anything from master to both 1.2.x and 1.3.x. nbd.

**Comment by jreback** [comment]
well only if we are actually patching 1.2,x (which at this point should be very very minimal)

**Comment by simonjayhawkins** [comment]
> very very minimal

sure. and autobackporting to 1.3.x from master after the fork at 1.3.0.rc0 should just work in more cases until the branches are significantly diverged.

if we can do 1.2.5 before 1.3.0.rc0 great, but we don't need to block 1.3.0.rc0 on 1.2.5

**Comment by jorisvandenbossche** [comment]
Note that https://github.com/pandas-dev/pandas/pull/41711 might have given a performance regression, which might be best to check first since that's backported to the 1.2.x branch.

**Comment by simonjayhawkins** [comment]
Thanks @jorisvandenbossche I think we were ready to go with 1.2.5 last week and then #41797 (and #41778 but that is now fixed) was reported and i've not got round to look at it and no PRs have been open for this, so 1.2.5 has been delayed anyway.

**Comment by jorisvandenbossche** [comment]
I have a small PR for fixing the performance regression: https://github.com/pandas-dev/pandas/pull/41920, so that should address this issue for 1.2.5.

**Comment by simonjayhawkins** [comment]
pre-release checklist

- ✅  wheels building https://github.com/MacPython/pandas-wheels/pull/133
- ✅  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
- ❌   release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.5.html
- ✅  ci checks https://github.com/pandas-dev/pandas/commits/1.2.x
- ✅  no manual backports outstanding https://github.com/pandas-dev/pandas/pulls?q=is%3Apr+label%3A%22Still+Needs+Manual+Backport%22+is%3Aclosed
- ❌   no open issues/PRs (except this one) https://github.com/pandas-dev/pandas/milestone/85


#41780 for the docs, #41924 to be merged and backported, #40799, #41596 and #41797 unresolved.

**Comment by simonjayhawkins** [comment]
@pandas-dev/pandas-core I'm going to try and replicate the issue with the release readiness scripts locally https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22 and then look at #41797 after if no-one else gets a chance before. So could maybe release 1.2.5 in the next couple of days?

**Comment by simonjayhawkins** [comment]
> I'm going to try and replicate the issue with the release readiness scripts locally https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22

couldn't isolate which dependency change actually broke the github workflow but changing from "conda build" to "conda-build" fixed. 

"conda build" gives a "[y/N]" prompt and does nothing.

**Comment by simonjayhawkins** [comment]
pre-release checklist

⬜  wheels building https://github.com/MacPython/pandas-wheels/pull/133
⬜  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
⬜   release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.2.5.html
⬜  ci checks https://github.com/pandas-dev/pandas/commits/1.2.x
⬜  no manual backports outstanding https://github.com/pandas-dev/pandas/pulls?q=is%3Apr+label%3A%22Still+Needs+Manual+Backport%22+is%3Aclosed
⬜   no open issues/PRs (except this one) https://github.com/pandas-dev/pandas/milestone/85

**Comment by simonjayhawkins** [comment]
all pre-release checks passed. closing down milestone and starting release shortly.

**Comment by simonjayhawkins** [comment]
- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag https://github.com/pandas-dev/pandas/tags
- [ ] open PR to build wheels https://github.com/MacPython/pandas-wheels/pull/xxx
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.2.5/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release - https://github.com/pandas-dev/pandas/releases
- [ ] upload wheels to PyPI https://pypi.org/project/pandas/
- [ ] change link for stable docs https://pandas.pydata.org/docs/
- [ ] ~~Start v1.2.6 release notes https://github.com/pandas-dev/pandas/pull/xxx~~
- [ ] conda-forge bot picked up on release https://github.com/conda-forge/pandas-feedstock/pull/xxx
- [ ] available on conda-forge https://anaconda.org/conda-forge/pandas/files
- [ ] announce
